runtime/compiler/ilgen/Walker.cpp

```diff
@@ -1672,6 +1672,15 @@ TR_J9ByteCodeIlGenerator::stashArgumentsForOSR(TR_J9ByteCode byteCode)
    TR::MethodSymbol *symbol = symRef->getSymbol()->castToMethodSymbol();
    int32_t numArgs = symbol->getMethod()->numberOfExplicitParameters() + (symbol->isStatic() ? 0 : 1);
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   // The method expects one more argument than we need to stashed, which is the appendix object
+   // from side table. Interpreter will load the object itself
```

Comment:
I think the comment you have in your commit message is a better description of this code.  Could you use something like that instead?
```
// invokehandle/invokedynamic bytecode implicitly push an object from side
// table as the last argument of the call. This is part of the behavior of
// the bytecode in OpenJDK MethodHandle implementation. The object should
// not be stashed because the interpreter will load the object itself.
//
```

Reply:
Thanks. I've put more detailed comment there